supplementary/alica_ros1/alica_tests/src/test/test_blackboard.cpp

```diff
@@ -72,5 +118,359 @@ TEST_F(TestBlackBoard, testNotInheritBlackboardFlag)
     EXPECT_FALSE(alica::test::Util::getBasicPlan(ae, 1692837668719979457, 0)->getInheritBlackboard());
 }
 
+TEST_F(TestBlackboard, testWithoutPlan)
+{
+    Blackboard bb;
+    LockedBlackboardRW bbrw = LockedBlackboardRW(bb);
+
+    // set default types
+    bbrw.set<int64_t>("val1", 14l);
+    bbrw.set<double>("val2", 15.3);
+    bbrw.set<std::string>("val3", "test");
+    bbrw.set<bool>("val4", true);
+
+    // get default types
+    EXPECT_EQ(bbrw.get<int64_t>("val1"), 14l);
+    EXPECT_EQ(bbrw.get<double>("val2"), 15.3);
+    EXPECT_EQ(bbrw.get<std::string>("val3"), "test");
+    EXPECT_EQ(bbrw.get<bool>("val4"), true);
+
+    // unknown types
+    bbrw.set<PlanStatus>("val5", PlanStatus::Success);
+    EXPECT_EQ(bbrw.get<PlanStatus>("val5"), PlanStatus::Success);
+}
+
+TEST_F(TestBlackboard, testMappingFromBool)
+{
+    EXPECT_TRUE((checkMapping<bool, bool>(BBType::BOOL, true, BBType::BOOL, true)));
+    EXPECT_TRUE((checkMapping<bool, int64_t>(BBType::BOOL, true, BBType::INT64, 1)));
+    EXPECT_TRUE((checkMapping<bool, double>(BBType::BOOL, true, BBType::DOUBLE, 1.0)));
```

Comment:
I don't think bool to double (or even int) conversations are something we really want to support

Reply:
A single line change in 292 will determine the kind of conversions we support. Right now I use `std::is_constructible`. But we could of course use our own custom trait if we want specific conversions to be allowed/disallowed.

Reply:
As mentioned in comment above, I'm not convinced there is value supporting any conversion other than int->double.  Especially narrowing conversions should not be supported imo

Reply:
Yeah, I did look for a way to prevent narrowing conversions (that also causes the compiler warnings which had to be suppressed) but it turned out to be quite tricky. 
Also bool -> double is not a narrowing conversion so maybe it needs to be dealt with separately.
Let me look into this a bit & see if I can come up with a suitable trait

Reply:
I found a solution. Basically I now call the set API here to do the mapping. This implies that mapping & set now support the same kind of type conversions which is desirable. Now, only conversions within the same "type category" is supported i.e. int8_t, int32_t (etc.) to int64_t, float to double etc. but not in between categories i.e. int -> double (we can add this later if we want to support it), bool -> double etc.
The one thing missing is unsigned integer support which should be easy enough to add. I'll work on that now